bot/telegram_bot.py

```diff
@@ -183,10 +199,81 @@ async def reset(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
         reset_content = message_text(update.message)
         self.openai.reset_chat_history(chat_id=chat_id, content=reset_content)
         await update.effective_message.reply_text(
+
             message_thread_id=get_thread_id(update),
             text=localized_text('reset_done', self.config['bot_language'])
         )
 
+    async def unsend(self, update: Update, context: CallbackContext):
+        """
+        Unsend the last message sent by the bot
+        """
+        if not await self.is_allowed(update, context):
```

Comment:
should be `if not is_allowed(update, context)`, as `is_allowed` is imported and no longer part of this class